src/service/db.py

```diff
@@ -43,14 +51,20 @@ def _to_song_key(self, songName: str):
         )
 
     @with_cache("SongServiceByDB::get_format_tags")
-    def get_format_tags(self, spreadsheetId: str = "") -> Mapping[str, str]:
-        if spreadsheetId == "":
-            spreadsheetId = self.defaultSpreadsheetId
+    def get_format_tags(self, group: str = "") -> Mapping[str, str]:
+        spreadsheetId = self.groupToSpreadsheetIds.get(group, self.defaultSpreadsheetId)
 
         return self.service.songTemplateDB.get_format_tags(spreadsheetId)
 
-    def save_song(self, song: Song, spreadsheetId: str = ""):
-        if spreadsheetId == "":
-            spreadsheetId = self.defaultSpreadsheetId
+    @with_cache("SongServiceByDB::get_all_format_tags")
+    def get_all_format_tags(self) -> Mapping[str, str]:
```

Comment:
Oops, missed this one, add this to the parent class as an abstract method and we're good to merge

Reply:
fixed